app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/agent/DateOfCommencementController.scala

```diff
@@ -0,0 +1,101 @@
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.utilities.ImplicitDateFormatter
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.play.language.LanguageUtils
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.agent.date_of_commencement
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.FormUtil._
+
+
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DateOfCommencementController @Inject()(mcc: MessagesControllerComponents,
+                                              incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,                                              authService: AuthService,
+                                              val languageUtils: LanguageUtils)
+                                             (implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc)
+  with I18nSupport with ImplicitDateFormatter {
+
+  def view(dateOfCommencementForm: Form[BusinessStartDate], isEditMode: Boolean)(implicit request: Request[AnyContent]): Html = {
+    date_of_commencement(
+      dateOfCommencementForm = dateOfCommencementForm,
+      postAction = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.agent.routes.DateOfCommencementController.submit(isEditMode),
+      isEditMode,
+      backUrl = backUrl(isEditMode)
+    )
+  }
+
+  def show(isEditMode: Boolean): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      incomeTaxSubscriptionConnector.getSelfEmployments[BusinessStartDate](DateOfCommencementController.businessStartDateKey).map {
+        case Right(businessStartDateData) => Ok(view(form.fill(businessStartDateData),isEditMode))
+        case error => throw new InternalServerException(error.toString)
+      }
+    }
+  }
+
+
+  def submit(isEditMode: Boolean): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      form.bindFromRequest.fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors,isEditMode))),
+        businessStartDateData =>
+          incomeTaxSubscriptionConnector.saveSelfEmployments(DateOfCommencementController.businessStartDateKey, businessStartDateData) map (_ =>
+            if (isEditMode) {
+              Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.agent.routes.DateOfCommencementController.show())
+            } else {
+              Redirect(appConfig.incomeTaxSubscriptionFrontendBaseUrl + "/client/business/name")
+            }
+            )
+      )
+    }
+  }
+
+  def backUrl(isEditMode: Boolean): String = {
+    if (isEditMode) {
+      uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.agent.routes.DateOfCommencementController.show().url
+    } else {
+      appConfig.incomeTaxSubscriptionFrontendBaseUrl + "/eligibility/sole-trader-start-date"
```

Comment:
Please redirect back to the income source selection page on the subscription frontend instead.